src/lib/__tests__/credentials.test.ts

```diff
@@ -0,0 +1,195 @@
+/* eslint-disable no-console */
+// @ts-ignore
+import { poll } from 'await-poll';
+import { Subject } from 'rxjs';
+import path from 'path';
+import indy from 'indy-sdk';
+import { Agent } from '..';
+import { SubjectInboundTransporter, SubjectOutboundTransporter } from './helpers';
+import { CredentialRecord } from '../storage/CredentialRecord';
+import { SchemaTemplate, CredDefTemplate } from '../agent/LedgerService';
+import { CredentialPreview } from '../protocols/credentials/messages/CredentialOfferMessage';
+import { CredentialState } from '../protocols/credentials/CredentialState';
+
+jest.setTimeout(15000);
+
+const faberConfig = {
+  label: 'Faber',
+  walletConfig: { id: 'credentials-test-faber' },
+  walletCredentials: { key: '00000000000000000000000000000Test01' },
+  publicDidSeed: process.env.TEST_AGENT_PUBLIC_DID_SEED,
+};
+
+const aliceConfig = {
+  label: 'Alice',
+  walletConfig: { id: 'credentials-test-alice' },
+  walletCredentials: { key: '00000000000000000000000000000Test01' },
+};
+
+const poolName = 'test-pool-credentials';
+const poolConfig = {
+  genesis_txn: process.env.GENESIS_TXN_PATH
+    ? path.resolve(process.env.GENESIS_TXN_PATH)
+    : path.join(__dirname, '../../../network/genesis/local-genesis.txn'),
+};
+
+const credentialPreview = new CredentialPreview({
+  attributes: [
+    {
+      name: 'name',
+      mimeType: 'text/plain',
+      value: 'John',
+    },
+    {
+      name: 'age',
+      mimeType: 'text/plain',
+      value: '99',
+    },
+  ],
+});
+
+describe('credentials', () => {
+  let faberAgent: Agent;
+  let aliceAgent: Agent;
+  let credDefId: CredDefId;
+
+  beforeAll(async () => {
+    const faberMessages = new Subject();
+    const aliceMessages = new Subject();
+
+    const faberAgentInbound = new SubjectInboundTransporter(faberMessages);
+    const faberAgentOutbound = new SubjectOutboundTransporter(aliceMessages);
+    const aliceAgentInbound = new SubjectInboundTransporter(aliceMessages);
+    const aliceAgentOutbound = new SubjectOutboundTransporter(faberMessages);
+
+    faberAgent = new Agent(faberConfig, faberAgentInbound, faberAgentOutbound, indy);
+    aliceAgent = new Agent(aliceConfig, aliceAgentInbound, aliceAgentOutbound, indy);
+    await faberAgent.init();
+    await aliceAgent.init();
+
+    console.log(`Connecting to ledger pool ${poolName} with config:`, poolConfig);
+    await faberAgent.ledger.connect(poolName, poolConfig);
+
+    const schemaTemplate = {
+      name: `test-schema-${Date.now()}`,
+      attributes: ['name', 'age'],
+      version: '1.0',
+    };
+    const [, ledgerSchema] = await registerSchema(faberAgent, schemaTemplate);
+
+    const definitionTemplate = {
+      schema: ledgerSchema,
```

Comment:
I'm curious. Is there a reason the definitionTemplate includes the schema instead of the schema id? I see indy requires the schema, but as the schema should already be registered an id seems sufficient.

Reply:
Good point. I don't know 🤷‍♂️  Do you suggest passing just `schemaId` into `definitionTemplate` and then get `schema` by calling `ledgerService.getCredentialSchema(schemaId)`? That would nice simplification, but not sure about doing this step right now. I would rather see the framework in real-world usage first and get some feedback.